<launch>
    <arg name="lidar_camera_fusion" default="true"/>
    <arg name="detection_groundtruth" default="false"/>

    <!-- Detection Groundtruth -->

    <group if="$(eval detection_groundtruth == true)">
         <node pkg="t4ac_smartmot_ros" type="t4ac_ab4cogt_ros_node.py" name="t4ac_ab4cogt_ros_node" output="screen"/>
    </group>
    <!--  -->

    <arg name="t4ac_smartmot_ros_node" value="/t4ac/perception/tracking_and_prediction/classic/t4ac_smartmot_ros/t4ac_smartmot_ros_node"/>
    <arg name="t4ac_sensor_fusion_ros_node" value="/t4ac/perception/detection/sensor_fusion/t4ac_sensor_fusion_ros/t4ac_sensor_fusion_ros_node"/>
    <arg name="t4ac_BEV_from_2D_detector_ros_node" value="/t4ac/perception/detection/sensor_fusion/t4ac_sensor_fusion_ros/t4ac_BEV_from_2D_detector_ros_node"/>
    
    <group ns="$(arg t4ac_smartmot_ros_node)">
        <param name="root" type="string" value="/t4ac/perception/tracking_and_prediction/classic/t4ac_smartmot_ros/t4ac_smartmot_ros_node" />
        <param name="display" type="bool" value="false"/> 
        <param name="trajectory_prediction" type="bool" value="true" />
        <param name="use_grid" type="bool" value="false" />
        <param name="node_name" type="string" value="t4ac_smartmot_ros_node" />

        <param name="pub_rectangular_monitorized_area_marker" type="string" value="$(arg t4ac_smartmot_ros_node)/rectangular_monitorized_area_marker"/>
        <param name="pub_particular_monitorized_areas_marker" type="string" value="$(arg t4ac_smartmot_ros_node)/particular_monitorized_areas_marker"/>
        <param name="pub_BEV_tracked_obstacles_marker" type="string" value="$(arg t4ac_smartmot_ros_node)/BEV_tracked_obstacles_marker"/>
        <param name="pub_ego_vehicle_forecasted_trajectory_marker" type="string" value="$(arg t4ac_smartmot_ros_node)/ego_vehicle_forecasted_trajectory_marker"/>
        <param name="pub_predicted_collision" type="string" value="$(arg t4ac_smartmot_ros_node)/predicted_collision"/> <!-- Unexpected Pedestrian or Collision -->
        <param name="pub_nearest_object_distance" type="string" value="$(arg t4ac_smartmot_ros_node)/nearest_object_distance"/> <!-- ACC or Obstacle in the route-->
        <param name="pub_pedestrian_crossing_occupied" type="string" value="$(arg t4ac_smartmot_ros_node)/pedestrian_crossing_occupied"/> <!-- Pedestrian Crossing -->
        <param name="pub_merge_occupied" type="string" value="$(arg t4ac_smartmot_ros_node)/merge_occupied"/> <!-- Give Way / Stop -->
        <param name="pub_front_obstacle" type="string" value="$(arg t4ac_smartmot_ros_node)/front_obstacle"/> <!-- Overtaking -->
        <param name="pub_left_lane_occupied" type="string" value="$(arg t4ac_smartmot_ros_node)/left_lane_occupied"/> <!-- Overtaking -->
        <param name="pub_right_lane_occupied" type="string" value="$(arg t4ac_smartmot_ros_node)/right_lane_occupied"/> <!-- Overtaking -->
        
        <!-- TODO: Improve this -->
        
        <group if="$(eval lidar_camera_fusion == true)">
            <param name="sub_BEV_merged_obstacles" type="string" value="$(arg t4ac_sensor_fusion_ros_node)/BEV_merged_obstacles"/>
            <group if="$(eval detection_groundtruth == true)">
                <param name="sub_BEV_merged_obstacles" type="string" value="/t4ac/perception/detection/BEV_groundtruth_obstacles"/>
            </group>
        </group>
        <group if="$(eval lidar_camera_fusion == false)">
            <param name="sub_BEV_merged_obstacles" type="string" value="$(arg t4ac_BEV_from_2D_detector_ros_node)/BEV_image_obstacles"/>
        </group>
        
        <param name="sub_localization_pose" type="string" value="/t4ac/localization/pose"/>
        <param name="sub_monitorized_lanes" type="string" value="/t4ac/perception/monitors/monitorized_lanes"/>

    </group>
    <node pkg="t4ac_smartmot_ros" type="t4ac_smartmot_ros_node.py" name="t4ac_smartmot_ros_node" output="screen"/>
</launch>

<!-- output="screen" -->

<!-- rosbag record -O "test_overtaking.bag" /carla/ego_vehicle/camera/rgb/front/camera_info /carla/ego_vehicle/camera/rgb/front/image_color /carla/ego_vehicle/camera/depth/front/image_depth /carla/ego_vehicle/vehicle_info /carla/objects /t4ac/localization/pose /t4ac/mapping/monitor/lanes /t4ac/mapping/monitor/lanes_marker /tf /carla/ego_vehicle/lidar/lidar1/point_cloud -->

